firmware/bittide-sys/src/gather_unit.rs

```diff
@@ -34,29 +36,30 @@ impl<const FRAME_SIZE: usize> GatherUnit<FRAME_SIZE> {
     ///
     /// The FDT must be a valid description of the hardware components that this
     /// code is running on.
-    pub unsafe fn from_fdt(fdt: &fdt::Fdt) -> Result<Self, FdtLoadError> {
+    pub unsafe fn from_fdt_node(node: &FdtNode) -> Result<Self, ComponentLoadError> {
         let get_node = |path| {
-            fdt.find_node(path)
-                .ok_or(FdtLoadError::FdtNodeNotFound(path))
+            node.children()
+                .find(|child| matches_fdt_name(child, path))
+                .ok_or(ComponentLoadError::FdtNodeNotFound(path))
         };
 
         let get_reg = |node: &fdt::node::FdtNode, component| {
             node.reg()
-                .ok_or(FdtLoadError::RegNotFound { component })?
+                .ok_or(ComponentLoadError::RegNotFound { component })?
                 .next()
-                .ok_or(FdtLoadError::RegNotFound { component })
+                .ok_or(ComponentLoadError::RegNotFound { component })
         };
 
-        let memory_node = get_node("/gather-unit/gather-memory")?;
-        let sequence_counter_node = get_node("/gather-unit/sequence-counter-reg")?;
-        let send_sequence_counter_node = get_node("/gather-unit/send-sequence-counter-reg")?;
-        let metacycle_register_node = get_node("/gather-unit/metacycle-reg")?;
+        let memory_node = get_node("gather-memory")?;
+        let sequence_counter_node = get_node("sequence-counter-reg")?;
+        let send_sequence_counter_node = get_node("send-sequence-counter-reg")?;
+        let metacycle_register_node = get_node("metacycle-reg")?;
 
```

Comment:
IMO we should be careful sacrificing code readability for performance reasons, especially if it is as trivial as this. For full system / PE simulation we should probably move to Verilator.

If it really is urgent to reduce runtime, move these strings to constants. E.g., in some constants file

```
GATHER_UNIT_MEMORY = "gum";
```

in this file:

```rust
get_node(GATHER_UNIT_MEMORY)?;
```

Reply:
Yeah I think so too. This is also initialisation code that only runs once, so at least when it comes to the overall performance of the system, this is not really critical code. For debugging it can be annoying, yeah, if this code is too slow for comfort we should definitely reach for constants to keep the code readable!